scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -91,203 +104,325 @@ def _national_1900_1979(file_path: str) -> pd.DataFrame:
-    if int(year) < 1960:
-        # schema is changing after 1940
-        if int(year) < 1940:
-            df = pd.read_csv(file_path,
-                             thousands=',',
-                             engine='python',
-                             skiprows=7,
-                             skipfooter=92,
-                             header=None)
+    try:
+        # extracting year from file path.
+        year = file_path[-8:-4]
+        # schema is changing after 1960
+        if int(year) < 1960:
+            # schema is changing after 1940
+            if int(year) < 1940:
+                df = pd.read_csv(file_path,
+                                 thousands=',',
+                                 engine='python',
+                                 skiprows=7,
+                                 skipfooter=92,
+                                 header=None)
+            else:
+                df = pd.read_csv(file_path,
+                                 thousands=',',
+                                 engine='python',
+                                 skiprows=7,
+                                 skipfooter=102,
+                                 header=None)
+            df.columns = [
+                'Age', 'Total', 'Count_Person_Male', 'Count_Person_Female',
+                'White Total', 'White Male', 'White Female', 'NonWhite Total',
+                'NonWhite Male', 'NonWhite Female'
+            ]
+            df = df.drop(columns=[
+                'Age', 'Total', 'White Total', 'White Male', 'White Female',
+                'NonWhite Total', 'NonWhite Male', 'NonWhite Female'
+            ])
+            df['Year'] = year
+            df.insert(0, 'geo_ID', 'country/USA', True)
+            df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
         else:
             df = pd.read_csv(file_path,
                              thousands=',',
                              engine='python',
-                             skiprows=7,
+                             skiprows=6,
                              skipfooter=102,
                              header=None)
+            df.columns = [
+                'Age', 'Total', 'Count_Person_Male', 'Count_Person_Female',
+                'White Total', 'White Male', 'White Female', 'Black Total',
+                'Black Male', 'Black Female', 'OtherRace Total',
+                'OtherRace Male', 'OtherRace Female'
+            ]
+            # dropping unwanted columns
+            df = df.drop(columns=[
+                'Age', 'Total', 'White Total', 'White Male', 'White Female',
+                'Black Total', 'Black Male', 'Black Female', 'OtherRace Total',
+                'OtherRace Male', 'OtherRace Female'
+            ])
+            # adding geoid, year and measurement method
+            df['Year'] = year
+            df.insert(0, 'geo_ID', 'country/USA', True)
+            df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
```

Comment:
This eems common to both if and else and can be moved out.

Reply:
It has been modified accordingly.